File: external/boost_1.55.0/libs/python/test/tuple.cpp
// Copyright David Abrahams 2005. Distributed under the Boost
// Software License, Version 1.0. (See accompanying
// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)

#include <boost/python/module.hpp>
#include <boost/python/def.hpp>
#include <boost/python/class.hpp>
#include <boost/python/tuple.hpp>

using namespace boost::python;

object convert_to_tuple(object data)
{
    return tuple(data);
}

void test_operators(tuple t1, tuple t2, object print)
{
    print(t1 + t2);
}

tuple mktuple0() { return make_tuple(); }
tuple mktuple1(int x) { return make_tuple(x); }
tuple mktuple2(char const* a1, int x) { return make_tuple(a1, x); }

BOOST_PYTHON_MODULE(tuple_ext)
{
    def("convert_to_tuple",convert_to_tuple);
    def("test_operators",test_operators);
    def("make_tuple", mktuple0);
    def("make_tuple", mktuple1);
    def("make_tuple", mktuple2);
}

File: external/boost_1.55.0/libs/python/test/tuple.py
# Copyright David Abrahams 2004. Distributed under the Boost
# Software License, Version 1.0. (See accompanying
# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
"""
>>> from tuple_ext import *
>>> def printer(*args):
...     for x in args: print x,
...     print
...
>>> print convert_to_tuple("this is a test string")
('t', 'h', 'i', 's', ' ', 'i', 's', ' ', 'a', ' ', 't', 'e', 's', 't', ' ', 's', 't', 'r', 'i', 'n', 'g')
>>> t1 = convert_to_tuple("this is")
>>> t2 = (1,2,3,4)
>>> test_operators(t1,t2,printer) #doctest: +NORMALIZE_WHITESPACE
('t', 'h', 'i', 's', ' ', 'i', 's', 1, 2, 3, 4)
>>> make_tuple()
()
>>> make_tuple(42)
(42,)
>>> make_tuple('hello', 42)
('hello', 42)
"""

def run(args = None):
    import sys
    import doctest

    if args is not None:
        sys.argv = args
    return doctest.testmod(sys.modules.get(__name__))
    
if __name__ == '__main__':
    print "running..."
    import sys
    status = run()[0]
    if (status == 0): print "Done."
    sys.exit(status)
